services/accounts/accounts.go

```diff
@@ -189,6 +239,49 @@ func (api *API) GenerateAccount(
 	return api.SaveAccounts(ctx, []accounts.Account{acc})
 }
 
+func (api *API) GenerateAccountWithDerivedPath(
```

Comment:
Same, can you call this function from generateAccount? to avoid code duplication